xagon-react-app/src/components/GameComponent/inputManager/FlipGesture.ts

```diff
@@ -0,0 +1,180 @@
+        if (data) {
+          assetMesh.metadata.vertices = data.vertices;
+          assetMesh.metadata.edges = data.edges;
+
+          this.firstTriangle = assetMesh;
+        }
+      }
+    }
+  }
+
+  public onMove(): void {
+    if (this.firstTriangle) {
+      const firstTriangle = this.firstTriangle.metadata.triangle;
+
+      const pickinfo = this.context.scene.pick(
+        this.context.scene.pointerX,
+        this.context.scene.pointerY,
+      );
+
+      if (
+        pickinfo?.pickedMesh &&
+        pickinfo?.pickedMesh?.metadata.triangle.triangleId !==
+          firstTriangle.triangleId
+      ) {
+        const mesh = pickinfo?.pickedMesh;
+        if (mesh) {
+          const assetMesh = getAssetMesh({
+            scene: this.context.scene,
+            triangleMesh: mesh,
+          });
+          if (assetMesh) {
+            const data = this.computeObjSpaceData(assetMesh);
+            if (data) {
+              assetMesh.metadata.vertices = data.vertices;
+              assetMesh.metadata.edges = data.edges;
+
+              this.secondTriangle = assetMesh;
+            }
+          }
+
+          if (this.secondTriangle) {
+            const secondTriangle = this.secondTriangle.metadata.triangle;
+
+            const scalingNodeFirstTriangle = this.firstTriangle.parent;
+            const flipNodeFirstTriangle = scalingNodeFirstTriangle?.parent;
+            const scalingNodeSecondTriangle = this.secondTriangle.parent;
+            const flipNodeSecondTriangle = scalingNodeSecondTriangle?.parent;
+
+            const firstEdges = this.firstTriangle.metadata.edges;
+            const secondEdges = this.secondTriangle.metadata.edges;
+            const firstVertices = this.firstTriangle.metadata.vertices;
+            const secondVertices = this.secondTriangle.metadata.vertices;
+
+            if (flipNodeFirstTriangle && flipNodeSecondTriangle) {
+              flipNodeFirstTriangle.position = Vector3.Center(
+                firstVertices[0],
+                firstVertices[1],
+              );
+              scalingNodeFirstTriangle.position = Vector3.Center(
+                firstVertices[0],
+                firstVertices[1],
+              ).scale(-1);
+              flipNodeSecondTriangle.position = Vector3.Center(
+                secondVertices[0],
+                secondVertices[2],
+              );
+              scalingNodeSecondTriangle.position = Vector3.Center(
+                secondVertices[0],
+                secondVertices[2],
+              ).scale(-1);
+
+              this.context.scene.registerBeforeRender(() => {
+                flipNodeFirstTriangle.rotate(firstEdges[0], Math.PI * 0.01);
+                flipNodeSecondTriangle.rotate(secondEdges[2], -Math.PI * 0.01);
+              });
+            }
+          }
+        }
+      }
+    }
+  }
+
+  public onRelease(pointerInfo: PointerInfo): void {
+    if (this.firstTriangle && this.secondTriangle) {
```

Comment:
immagino questo codice sia di debug/test.. magari metti un commento per non dimenticarti poi cose in giro